.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,28 @@
+name: Pull Request Action with Caching
+
+on: pull_request
```

Comment:
Is there a reason to have this check only on a pull request?

Reply:
I could do it on push instead. The idea was to keep the number of builds low to remain within free tier. I can see a situation though were you push to a branch which is already at the pull request stage. 

I think upon reflection, push sounds better?

Reply:
msg me when this is resolved to your liking and i can merge